apps/heft/src/plugins/TypeScriptPlugin/TypeScriptBuilder.ts

```diff
@@ -121,13 +122,33 @@ export class TypeScriptBuilder extends SubprocessRunnerBase<ITypeScriptBuilderCo
   private _typescriptTerminal!: Terminal;
   private _emitCompletedCallbackManager: EmitCompletedCallbackManager;
 
+  private __tsCacheFilePath!: string;
   private _tsReadJsonCache: Map<string, object> = new Map<string, object>();
   private _cachedFileSystem: TypeScriptCachedFileSystem = new TypeScriptCachedFileSystem();
 
   public get filename(): string {
     return __filename;
   }
 
+  private get _tsCacheFilePath(): string {
+    if (!this.__tsCacheFilePath) {
+      const configHash: crypto.Hash = Tslint.getConfigHash(
+        this._configuration.tsconfigPath,
+        this._typescriptTerminal,
+        this._cachedFileSystem
+      );
+      configHash.update(JSON.stringify(this._configuration.additionalModuleKindsToEmit || {}));
+      const serializedConfigHash: string = configHash.digest('hex');
+
+      this.__tsCacheFilePath = path.posix.join(
+        this._configuration.buildMetadataFolder,
+        `ts_${serializedConfigHash}.json`
+      );
```

Comment:
Should just be able to say

```suggestion
      this.__tsCacheFilePath = `${this._configuration.buildMetadataFolder}/ts_${serializedConfigHash}.json`;
```

right?